common/k8s/src/event_source.rs

```diff
@@ -0,0 +1,226 @@
+                    .flatten(),
+                count,
+            },
+        }
+    }
+}
+
+// TODO test from...
+
+#[derive(Serialize, Debug)]
+struct EventLogLine {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    _name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    level: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    message: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    _app: Option<String>,
+    kube: EventLogLineInner,
+}
+
+#[derive(Serialize, Debug)]
+struct EventLogLineInner {
+    type_: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    action: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    resource: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    namespace: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    reason: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    message: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    component: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    node: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    first_time: Option<Time>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    time: Option<Time>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    age: Option<i32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    count: Option<i32>,
+}
+
+pub struct K8sEventStream {
+    api: Api<Event>,
+    node_name: Option<String>,
+}
+
+impl K8sEventStream {
+    pub fn new(config: kube::Config, node_name: Option<String>) -> Result<Self, K8sError> {
+        Ok(Self {
+            api: Api::<Event>::all(Client::new(config)),
+            node_name,
+        })
+    }
+
+    pub fn try_default(node_name: Option<String>) -> Result<Self, K8sError> {
+        let config = match Config::from_cluster_env() {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(K8sError::InitializationError(format!(
+                    "unable to get cluster configuration info: {}",
+                    e
+                )))
+            }
+        };
+        Self::new(config, node_name)
+    }
+
+    pub fn event_stream(&self) -> impl Stream<Item = Vec<LineBuilder>> + Send {
+        let start_stream = {
+            let node_name = self.node_name.clone();
+            let api = self.api.clone();
+            move || {
+                let mut params = ListParams::default();
+                if let Some(ref node_name) = node_name {
+                    params.field_selector = Some(format!("source.host=={}", node_name));
+                } else {
+                    info!("Node name not available, logging all cluster events, will cause duplication per node.")
+                }
```

Comment:
I believe this still needs to be changed because you can't rely on `source.host` as a field selector as you've mentioned before?